temporal-sdk/src/main/java/io/temporal/failure/FailureConverter.java

```diff
@@ -195,11 +190,14 @@ public static Failure exceptionToFailureNoUnwrapping(Throwable e) {
     } else {
       message = e.getMessage() == null ? "" : e.getMessage();
     }
+    if (e instanceof CheckedExceptionWrapper) {
+      return exceptionToFailure(e.getCause());
+    }
```

Comment:
I would move it to the top of the method to avoid unnecessary e.getMessage call.

Reply:
Done